packages/esm-patient-notes-app/src/notes/visit-notes-form.component.tsx

```diff
@@ -32,68 +31,42 @@ import {
-  const [locationUuid, setLocationUuid] = React.useState<string | null>(null);
-  const [providerUuid, setProviderUuid] = React.useState<string | null>(null);
-  const [primarySearchTerm, setPrimarySearchTerm] = React.useState<string | null>('');
-  const [secondarySearchTerm, setSecondarySearchTerm] = React.useState<string | null>('');
-  const [selectedPrimaryDiagnoses, setSelectedPrimaryDiagnoses] = React.useState<Array<Diagnosis>>([]);
-  const [selectedSecondaryDiagnoses, setSelectedSecondaryDiagnoses] = React.useState<Array<Diagnosis>>([]);
-  const [searchPrimaryResults, setSearchPrimaryResults] = React.useState<null | Array<Concept>>(null);
-  const [searchSecondaryResults, setSearchSecondaryResults] = React.useState<null | Array<Concept>>(null);
-  const [combinedDiagnoses, setCombinedDiagnoses] = React.useState<Array<Diagnosis>>([]);
-  const [visitDateTime, setVisitDateTime] = React.useState(new Date());
-
-  React.useEffect(() => {
-    if (session && !currentSessionLocationUuid && !currentSessionProviderUuid) {
-      setCurrentSessionLocationUuid(session?.sessionLocation?.uuid);
-      setCurrentSessionProviderUuid(session?.currentProvider?.uuid);
-    }
-  }, [currentSessionLocationUuid, currentSessionProviderUuid, session]);
+  const [clinicalNote, setClinicalNote] = useState('');
+  const [isPrimarySearching, setIsPrimarySearching] = useState(false);
+  const [isSecondarySearching, setIsSecondarySearching] = useState(false);
+  const [isHandlingSubmit, setIsHandlingSubmit] = useState(false);
+  const [primarySearchTerm, setPrimarySearchTerm] = useState<string | null>('');
+  const [secondarySearchTerm, setSecondarySearchTerm] = useState<string | null>('');
+  const [selectedPrimaryDiagnoses, setSelectedPrimaryDiagnoses] = useState<Array<Diagnosis>>([]);
+  const [selectedSecondaryDiagnoses, setSelectedSecondaryDiagnoses] = useState<Array<Diagnosis>>([]);
+  const [searchPrimaryResults, setSearchPrimaryResults] = useState<null | Array<Concept>>(null);
+  const [searchSecondaryResults, setSearchSecondaryResults] = useState<null | Array<Concept>>(null);
+  const [combinedDiagnoses, setCombinedDiagnoses] = useState<Array<Diagnosis>>([]);
+  const [visitDateTime, setVisitDateTime] = useState(new Date());
 
-  React.useEffect(() => {
-    const ac = new AbortController();
-    if (currentSessionProviderUuid) {
-      fetchProviderByUuid(ac, currentSessionProviderUuid).then(({ data }) => {
-        setProviderUuid(data.uuid);
-      });
-    }
-    if (currentSessionLocationUuid) {
-      fetchLocationByUuid(ac, currentSessionLocationUuid).then(({ data }) => setLocationUuid(data.uuid));
-    }
-  }, [currentSessionLocationUuid, currentSessionProviderUuid]);
+  const { mutateVisitNotes } = useVisitNotes(patientUuid);
+  const { locationUuid } = useLocationUuid(session?.sessionLocation?.uuid);
+  const { providerUuid } = useProviderUuid(session?.currentProvider?.uuid);
```

Comment:
I don't know that these hooks are useful... I mean, I don't really see the point of the code they're replacing either... If we load these UUIDs from the session endpoint, those UUIDs willl always be valid so the re-checking with the backend is a a couple of useless calls. I'd just go with:

```
const locationUuid = session?.sessionLocation?.uuid;
const providerUuuid = session?.currentProvider?.uuid;
```

Reply:
Great catch. Felt icky writing this.